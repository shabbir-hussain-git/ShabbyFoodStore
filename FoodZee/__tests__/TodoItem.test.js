import 'react-native';
import React from 'react';

import {it} from '@jest/globals';

// import renderer, {fireEvent, getByText} from 'react-test-renderer';
import TodoItem from '../components/TodoItem';

import {render, screen, fireEvent} from '@testing-library/react-native';
it('renders correctly', () => {
  const x = {
    todo: {
      todoText: 'Go To Gym',
    },
  };
  const t1 = jest.fn();
  const t2 = jest.fn();
  x.editTodoItem = t1;
  x.deleteTodoItem = t2;

  // const tree = renderer.create(<TodoItem {...x} />).toJSON();
  const tree = render(<TodoItem {...x} />);
  const btn1 = screen.getByTestId('btn1');
  const ans = screen.getByTestId('ans');
  fireEvent.press(btn1, {todo: {todoText: 'Hello'}});
  expect(ans.children[0]).toBe('No Hello');

  expect(tree).toMatchSnapshot();
});
