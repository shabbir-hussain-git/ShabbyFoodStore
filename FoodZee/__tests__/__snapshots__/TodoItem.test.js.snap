// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  className="todoItem"
>
  <View
    className="todoItemText"
  >
    <Text>
      Go To Gym
    </Text>
  </View>
  <View
    className="todoItemText"
  >
    <Text
      testID="ans"
    >
      No Hello
    </Text>
  </View>
  <View
    className="todoItemControls"
  >
    <View
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="btn1"
    >
      <Text>
        Edit
      </Text>
    </View>
    <View
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text>
        Del
      </Text>
    </View>
  </View>
</View>
`;
